pkg/services/ngalert/ngalert.go

```diff
@@ -151,7 +137,32 @@ func (ng *AlertNG) init() error {
 		appUrl = nil
 	}
 
-	stateManager := state.NewManager(ng.Log, ng.Metrics.GetStateMetrics(), appUrl, store, store, ng.dashboardService, ng.imageService, clock.New())
+	clk := clock.New()
+
+	alertsRouter := sender.NewAlertsRouter(ng.MultiOrgAlertmanager, store, clk, appUrl, ng.Cfg.UnifiedAlerting.DisabledOrgs, ng.Cfg.UnifiedAlerting.AdminConfigPollInterval)
+
+	if err := alertsRouter.SyncAndApplyConfigFromDatabase(); err != nil {
+		return fmt.Errorf("failed to initialize alerting because alert notifications router failed to warm up: %w", err)
+	}
+
+	ng.AlertsRouter = alertsRouter
+
+	schedCfg := schedule.SchedulerCfg{
+		C:               clk,
+		BaseInterval:    ng.Cfg.UnifiedAlerting.BaseInterval,
+		Logger:          ng.Log,
+		MaxAttempts:     ng.Cfg.UnifiedAlerting.MaxAttempts,
+		Evaluator:       eval.NewEvaluator(ng.Cfg, ng.Log, ng.DataSourceCache, ng.SecretsService, ng.ExpressionService),
+		InstanceStore:   store,
+		RuleStore:       store,
+		OrgStore:        store,
+		Metrics:         ng.Metrics.GetSchedulerMetrics(),
+		DisabledOrgs:    ng.Cfg.UnifiedAlerting.DisabledOrgs,
+		MinRuleInterval: ng.Cfg.UnifiedAlerting.MinInterval,
+		AlertSender:     alertsRouter,
```

Comment:
The AlertsSender interface is declared on scheduler side. I chose its name following the same conception like io.Reader and io.Writer interfaces. Agree that 'notifier' can be renamed to sender.

Reply:
I got tricked by the GitHub diff and it's not the `stateManager` is the `API` where it is called a router.

Reply:
> The AlertsSender interface is declared on scheduler side.

This is not true. Within `router.go`, I see `Senders map[int64]*Sender` so now we have:

`AlertsSender` in the schedule package
`AlertsRouter` that has multiple `Senders` in the `sender` package. 

In my eyes, if `router -> sender` are part of the internals of `sender` package and the scheduler _only_ interacts with `router` then it should be `router` *everywhere*.

What I want to avoid is the confusion that when we talk about the "alerts sender" we now need to ask - the sender in sender package or the sender in the schedule package?

Reply:
yes, the `Senders` field is exported now but it will be hidden in the package later when unit tests are updated, and it as well as the struct `Sender` will not be exported anywhere. 
As an alternative, I can think of `notifier` however, it also conflicts with `notifier` package where it represents a `contact point`. I would rather rename `Sender` that is in the `sender` package as well as the package itself to reflect that this is a client for external Alertmanagers. I think it will make more sense if we merge `sender` and `notifier` packages under the same package, e.g. `notifications`. 
I also do not think that this is a good idea to name the interface AlertsRouter as it will interfere with the struct itself.